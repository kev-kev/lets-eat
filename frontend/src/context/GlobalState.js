import React, { createContext, useReducer } from "react";
import { AppReducer } from "./AppReducer";
import { startOfWeek, differenceInDays, parseISO } from "date-fns";

const initialState = {
  user: null,
  isLoggingIn: false,
  isSubmittingRecipe: false,
  isFetchingRecipes: false,
  isFetchingRecipeList: false,
  isEditing: false,
  errors: {
    login: null,
    submit: null,
    grid: null,
    inbox: null,
  },
  selectedWeek: startOfWeek(new Date()),
  groceryList: null,
  favoritedRecipes: [],
  weeklyRecipes: [],
  otherRecipes: [],
  inboxRecipes: [],
  pendingRecipes: [],
  rejectedRecipes: [],
};
const rootURL = process.env.REACT_APP_API_URL;

const handleErrors = (response) => {
  if (!response.ok) throw Error(response.statusText);
  return response;
};

const isWeeklyRecipe = (recipeWeeks, selectedWeek) => {
  return recipeWeeks.some(
    (week) => differenceInDays(parseISO(week), selectedWeek) === 0
  );
};

export const GlobalContext = createContext(initialState);
export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(AppReducer, initialState);

  const getRecipesFromType = (type) => {
    //eslint-disable-next-line
    switch (type) {
      case "weekly":
        return [...state.weeklyRecipes];
      case "other":
        return [...state.otherRecipes];
      case "favorited":
        return [...state.favoritedRecipes];
      case "inbox":
        return [...state.inboxRecipes];
      case "pending":
        return [...state.pendingRecipes];
      case "rejected":
        return [...state.rejectedRecipes];
    }
  };

  const fetchRecipes = (user) => {
    dispatch({
      type: "FETCH_RECIPES",
    });
    fetch(rootURL + `/recipes/`)
      .then(handleErrors)
      .then((r) => r.json())
      .then((data) => {
        filterAndSetRecipes(data.recipes, user);
        dispatch({
          type: "FETCH_RECIPES_SUCCESS",
        });
      })
      .catch((error) => {
        dispatch({
          type: "FETCH_RECIPES_FAILURE",
          payload: error,
        });
      });
  };

  const editRecipe = (recipe) => {
    const subType = isWeeklyRecipe(recipe.weeks, state.selectedWeek)
      ? "weekly"
      : "other";
    fetch(rootURL + `/recipes/${recipe.id}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(recipe),
    })
      .then(handleErrors)
      .then(() => {
        const updatedRecipes = getRecipesFromType(subType);
        const recipeIndex = updatedRecipes.findIndex(
          (target) => target.id === recipe.id
        );
        updatedRecipes[recipeIndex] = recipe;

        dispatch({
          type: "EDIT_RECIPE_SUCCESS",
          payload: {
            updatedRecipes: updatedRecipes,
            type: subType,
          },
        });
      })
      .catch((error) => {
        dispatch({
          type: "EDIT_RECIPE_FAILURE",
          payload: error,
        });
      });
  };

  const filterAndSetRecipes = (recipes, user) => {
    const weeklyRecipes = [];
    const otherRecipes = [];
    const favoritedRecipes = [];
    const inboxRecipes = [];
    const pendingRecipes = [];
    const rejectedRecipes = [];

    recipes.forEach((recipe) => {
      //eslint-disable-next-line
      switch (recipe.status) {
        case "approved":
          if (isWeeklyRecipe(recipe.weeks, state.selectedWeek))
            weeklyRecipes.push(recipe);
          else otherRecipes.push(recipe);
          if (recipe.isFavorited) favoritedRecipes.push(recipe);
          break;
        case "pending":
          recipe.submittedBy === user.username
            ? pendingRecipes.push(recipe)
            : inboxRecipes.push(recipe);
          break;
        case "rejected":
          rejectedRecipes.push(recipe);
          break;
      }
    });

    const allRecipes = [
      ["weekly", weeklyRecipes],
      ["other", otherRecipes],
      ["favorited", favoritedRecipes],
      ["inbox", inboxRecipes],
      ["pending", pendingRecipes],
      ["rejected", rejectedRecipes],
    ];
    for (const subArr of allRecipes) {
      dispatch({
        type: "SET_RECIPES",
        payload: subArr,
      });
    }
  };

  function loginUser(username, password) {
    dispatch({
      type: "LOGIN_USER",
    });
    fetch(rootURL + "/login/", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ user: { username, password } }),
    })
      .then(handleErrors)
      .then((r) => r.json())
      .then((data) => {
        localStorage.setItem("token", data.jwt);
        dispatch({
          type: "LOGIN_SUCCESS",
          payload: {
            user: data.user,
          },
        });
      })
      .catch((error) => {
        dispatch({
          type: "LOGIN_FAILURE",
          payload: error,
        });
      });
  }

  function submitRecipe(name, link, notes, imgUrl, ingredients) {
    dispatch({
      type: "SUBMITTING_RECIPE",
    });
    fetch(rootURL + "/submit/", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        recipe: {
          name: name,
          link: link,
          notes: notes,
          user_id: state.user.id,
          img_url: imgUrl,
          ingredients: ingredients,
        },
      }),
    })
      .then(handleErrors)
      .then((r) => r.json())
      .then((data) => {
        dispatch({
          type: "SUBMIT_RECIPE_SUCCESS",
          payload: data.recipe,
        });
      })
      .catch((error) => {
        dispatch({
          type: "SUBMIT_RECIPE_FAILURE",
          payload: error,
        });
      });
  }

  function deleteRecipe(recipe_id) {
    fetch(rootURL + `/recipes/${recipe_id}`, {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        recipe: {
          id: recipe_id,
        },
      }),
    })
      .then(handleErrors)
      .then((r) => r.json())
      .then((data) => {
        dispatch({
          type: "DELETE_RECIPE_SUCCESS",
          payload: data.recipes.filter(
            (recipe) => recipe.status === "approved"
          ),
        });
      })
      .catch((error) => {
        dispatch({
          type: "DELETE_RECIPE_FAILURE",
          payload: error,
        });
      });
  }

  function changeRecipeStatus(recipe_id, recipe_status) {
    fetch(rootURL + `/recipes/${recipe_id}/`, {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        recipe: {
          status: recipe_status,
        },
      }),
    })
      .then(handleErrors)
      .then(() => {
        dispatch({
          type: "STATUS_UPDATE_SUCCESS",
        });
      })
      .catch((error) => {
        dispatch({
          type: "STATUS_UPDATE_FAILURE",
          payload: error,
        });
      });
  }

  function toggleFavorite(recipe_id, value) {
    fetch(rootURL + `/recipes/${recipe_id}`, {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        recipe: {
          is_favorited: value,
          id: recipe_id,
        },
      }),
    })
      .then(handleErrors)
      .then((r) => {
        if (r.status === 200)
          dispatch({
            type: "FAVORITE_UPDATE_SUCCESS",
            payload: { recipe_id, value },
          });
      })
      .catch((error) => {
        dispatch({
          type: "FAVORITE_UPDATE_FAILURE",
          payload: error,
        });
      });
  }

  function setWeeks(recipe_id, weeks) {
    fetch(rootURL + `/recipes/${recipe_id}`, {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        recipe: {
          weeks: weeks,
        },
      }),
    })
      .then(handleErrors)
      .then((r) => {
        if (r.status === 200)
          dispatch({
            type: "WEEKS_UPDATE_SUCCESS",
            payload: { recipe_id, weeks },
          });
      })
      .catch((error) => {
        dispatch({
          type: "WEEKS_UPDATE_FAILURE",
          payload: error,
        });
      });
  }

  function getGroceryList() {
    fetch(rootURL + `/grocery_list/?week=${state.selectedWeek}`)
      .then(handleErrors)
      .then((r) => r.json())
      .then((data) => {
        dispatch({
          type: "GET_GROCERY_LIST_SUCCESS",
          payload: data.groceryList,
        });
      })
      .catch((error) => {
        dispatch({
          type: "GET_GROCERY_LIST_FAILURE",
          payload: error,
        });
      });
  }

  function changeSelectedWeek(week) {
    dispatch({
      type: "CHANGE_SELECTED_WEEK",
      payload: week,
    });
  }

  function logoutUser() {
    dispatch({
      type: "LOGOUT_USER",
    });
  }

  function clearErrors() {
    dispatch({
      type: "CLEAR_ERRORS",
    });
  }

  return (
    <GlobalContext.Provider
      value={{
        user: state.user,
        isLoggingIn: state.isLoggingIn,
        isSubmittingRecipe: state.isSubmittingRecipe,
        login: loginUser,
        recipes: state.recipes,
        submitRecipe,
        deleteRecipe,
        changeRecipeStatus,
        logoutUser,
        errors: state.errors,
        clearErrors,
        isFetchingRecipes: state.isFetchingRecipes,
        isFetchingRecipeList: state.isFetchingRecipeList,
        selectedWeek: state.selectedWeek,
        changeSelectedWeek,
        setWeeks,
        toggleFavorite,
        getGroceryList,
        groceryList: state.groceryList,
        fetchRecipes,
        editRecipe,
        favoritedRecipes: state.favoritedRecipes,
        weeklyRecipes: state.weeklyRecipes,
        otherRecipes: state.otherRecipes,
        inboxRecipes: state.inboxRecipes,
        pendingRecipes: state.pendingRecipes,
        rejectedRecipes: state.rejectedRecipes,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
