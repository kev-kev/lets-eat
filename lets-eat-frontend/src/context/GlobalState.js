import React, { createContext, useReducer } from "react";
import AppReducer from "./AppReducer";

const initialState = {
  user: null,
  isLoggingIn: false
};

export const GlobalContext = createContext(initialState);
export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(AppReducer, initialState);

  function loginUser(username, password) {
    dispatch({
      type: "LOGIN_USER",
      payload: {
        isLoggingIn: true
      }
    });
    fetch("http://localhost:4000/login", {
      method: "POST",
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({user: {username, password}})
    })
      .then(r => {
        if (r.ok) {
          console.log(r)
          return r.json()
        } else {
          return dispatch({
            type: "LOGIN_FAILED",
            payload: {
              error: "Login failed"
            }
          })
        }
      })
      .then(data => {
        if (data.error) {
          console.log(data.error)
        } else {
        localStorage.setItem('token', data.jwt)
        dispatch({
        type: "LOGIN_SUCCESS",
        payload: {
          user: data.user
        }
      })}
    })

      
    // dispatch({
    //    type: "LOGIN_SUCCESS",
    //    payload: {
    //        user: {
    //            username: "mirmir"
    //        }
    //    }
    // });
    // Else:
    // dispatch({
    //    type: "LOGIN_FAILED",
    //    payload: {error}
    // })
  }

  return (
    <GlobalContext.Provider
      value={{
          user: state.user,
          isLoggingIn: state.isLoggingIn,
          login: loginUser
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};