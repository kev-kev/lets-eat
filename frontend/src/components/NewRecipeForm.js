import React, { useContext, useState, useEffect } from "react";
import {
  Button,
  CssBaseline,
  TextField,
  Typography,
  Container,
  CircularProgress,
  Snackbar,
} from "@material-ui/core/";
import { newRecipeFormStyle } from "../muiStyling";
import { GlobalContext } from "../context/GlobalState";
import { Alert } from "@material-ui/lab/";
import IngredientInputForm from "./IngredientInputForm";

const successMessage = "ヽ(*・ω・)ﾉ   Recipe Submitted!   ～('▽^人)";
const errorMessage = "Submission Failed (っ´ω`)ﾉ (╥ω╥)";

const NewRecipeForm = () => {
  const classes = newRecipeFormStyle();
  const { submitRecipe, isSubmittingRecipe, errors, clearErrors } =
    useContext(GlobalContext);

  const [title, setTitle] = useState("");
  const [imgUrl, setImgUrl] = useState("");
  const [link, setLink] = useState("");
  const [notes, setNotes] = useState("");
  const [ingredients, setIngredients] = useState([]);
  const [successSnackbar, setSuccessSnackbar] = useState(false);
  const [errorSnackbar, setErrorSnackbar] = useState(false);

  useEffect(() => {
    if (errors.submit) setErrorSnackbar(true);
  }, [errors.submit]);

  const handleClose = () => {
    setErrorSnackbar(false);
    setSuccessSnackbar(false);
    clearErrors();
  };

  const handleSubmit = (title, link) => {
    if (title === "" || link === "") {
      setSuccessSnackbar(false);
      setErrorSnackbar(true);
    } else {
      submitRecipe(title, link, notes, imgUrl, ingredients);
    }
  };

  const handleAddIngredientInput = () => {
    setIngredients([...ingredients, { name: "", count: 0, unit: "" }]);
  };

  const setAttributeForIngredient = (index, attribute, value) => {
    const updatedIngredients = [...ingredients];
    updatedIngredients[index][attribute] = value;
    setIngredients(updatedIngredients);
  };

  const handleDeleteIngredient = (index) => {
    const updatedIngredients = [...ingredients];
    updatedIngredients.splice(index, 1);
    setIngredients(updatedIngredients);
  };

  const renderIngredients = () => {
    return ingredients.map((ingredientInput, index) => {
      return (
        <IngredientInputForm
          ingredientInput={ingredientInput}
          setName={(ingredientName) =>
            setAttributeForIngredient(index, "name", ingredientName)
          }
          setCount={(count) => setAttributeForIngredient(index, "count", count)}
          setUnit={(unit) => setAttributeForIngredient(index, "unit", unit)}
          key={index}
          index={index}
          handleDeleteIngredient={(index) => handleDeleteIngredient(index)}
        />
      );
    });
  };

  if (isSubmittingRecipe) {
    return (
      <div className={classes.paper}>
        <CircularProgress className={classes.loading} />
      </div>
    );
  } else {
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Snackbar open={errorSnackbar} onClose={handleClose}>
          <Alert onClose={handleClose} severity="error">
            {errorMessage}
          </Alert>
        </Snackbar>
        <Snackbar open={successSnackbar} onClose={handleClose}>
          <Alert onClose={handleClose} severity="success">
            {successMessage}
          </Alert>
        </Snackbar>
        <div className={classes.paper}>
          <Typography component="h1" variant="h5">
            submit a new recipe idea
          </Typography>
          <form className={classes.form} noValidate>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="title"
              label="Recipe Title"
              name="title"
              autoComplete="title"
              autoFocus
              onChange={(e) => setTitle(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              fullWidth
              name="imgUrl"
              label="Image URL"
              type="imgUrl"
              id="imgUrl"
              autoComplete="current-imgUrl"
              onChange={(e) => setImgUrl(e.target.value)}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="link"
              label="Link to recipe"
              type="link"
              id="link"
              autoComplete="current-link"
              onChange={(e) => setLink(e.target.value)}
            />
            <Button
              color="primary"
              variant="contained"
              onClick={handleAddIngredientInput}
              className={classes.button}
            >
              New Ingredient
            </Button>
            <br />
            {renderIngredients()}
            <TextField
              className="textAreas"
              variant="outlined"
              margin="normal"
              multiline
              fullWidth
              rows={4}
              name="notes"
              label="Notes"
              type="notes"
              id="notes"
              autoComplete="notes"
              onChange={(e) => setNotes(e.target.value)}
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit + " " + classes.button}
              onClick={(e) => {
                e.preventDefault();
                handleSubmit(title, link);
              }}
            >
              submit
            </Button>
          </form>
        </div>
      </Container>
    );
  }
};

export default NewRecipeForm;
