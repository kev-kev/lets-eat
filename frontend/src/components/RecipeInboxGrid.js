import React, { useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import RecipeCard from './RecipeCard';
import { GlobalContext } from '../context/GlobalState';

const renderRecipeCard = (recipe, currentUser) => {
  if (recipe.submittedBy !== currentUser.username) {
    return (
      <Grid item xs key={recipe.name} >
        <RecipeCard
          id={recipe.id}
          isRecipeVoteCard={true}
          name={recipe.name}
          imgUrl={recipe.imgUrl}
          link={recipe.link}
          submittedBy={recipe.submittedBy} />
      </Grid>
    );
  }
}

const renderRecipeGrid = (recipes, currentUser) => {
  const pendingRecipes = recipes.filter(recipe => recipe.status === "pending")
  return pendingRecipes.map(recipe => {
      return renderRecipeCard(recipe, currentUser);
  });
}

export default function RecipeGrid() {
  const { recipes, user } = useContext(GlobalContext)
  return (
    <Grid 
      direction="row"
      justify="flex-start"
      alignItems="flex-start"
      spacing={2}
      container>
        {renderRecipeGrid(recipes, user)}
    </Grid>
  );
}