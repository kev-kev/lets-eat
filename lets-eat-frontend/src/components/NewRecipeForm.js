import React, { useContext, useState } from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import { GlobalContext } from "../context/GlobalState";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  kirby: {
    width: 'auto',
    height: theme.spacing(12),
    margin: '20px'
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export default function NewRecipeForm() {
  const classes = useStyles();
  const { submitRecipe } = useContext(GlobalContext);

  const [title, setTitle] = useState('')
  const [imgUrl, setImgUrl] = useState('')
  const [link, setLink] = useState('')
  const [notes, setNotes] = useState('')

  return (
      <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
          <Typography component="h1" variant="h5">
          submit a new recipe idea
          </Typography>
          <form className={classes.form} noValidate>
          <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="title"
              label="Recipe Title"
              name="title"
              autoComplete="title"
              autoFocus
              onChange = {e => setTitle(e.target.value)}
          />
          <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="imgUrl"
              label="Image URL"
              type="imgUrl"
              id="imgUrl"
              autoComplete="current-imgUrl"
              onChange = {e => setImgUrl(e.target.value)}

          />
          <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="link"
              label="Link to recipe"
              type="link"
              id="link"
              autoComplete="current-link"
              onChange = {e => setLink(e.target.value)}

          />
          <TextField
              variant="outlined"
              margin="normal"
              required
              multiline
              fullWidth
              rows={4}
              name="notes"
              label="Notes"
              type="notes"
              id="notes"
              autoComplete="notes"
              onChange = {e => setNotes(e.target.value)}
          />
          <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              onClick={e => {
                  e.preventDefault();
                  submitRecipe(title, link, notes)
              }}
          >
              submit
          </Button>
          </form>
      </div>
      </Container>
  );
}