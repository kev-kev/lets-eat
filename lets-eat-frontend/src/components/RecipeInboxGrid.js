import React, { useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import RecipeCard from './RecipeCard';
import { GlobalContext } from '../context/GlobalState';


const renderRecipeCard = (recipe) => {
    return (
        <Grid item xs={12} sm={6} md={3} xl={2} key={recipe.name} >
          <RecipeCard
            isRecipeVoteCard={true}
            name={recipe.name}
            imgUrl={recipe.img_url}
            link={recipe.link}
            submittedBy={recipe.submittedBy} />
        </Grid>
    );
}

const renderRecipeGrid = (recipes) => {
  const pendingRecipes = recipes.filter(recipe => recipe.status === "pending")
    return pendingRecipes.map(recipe => {
        return renderRecipeCard(recipe);
    });
}

export default function RecipeGrid() {
  const { recipes } = useContext(GlobalContext)
  return (
    <Grid 
      direction="row"
      justify="flex-start"
      alignItems="flex-start"
      spacing={3}
      container>
        {renderRecipeGrid(recipes)}
    </Grid>
  );
}