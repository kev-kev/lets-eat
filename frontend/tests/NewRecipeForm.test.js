import React from 'react'
import { render } from "@testing-library/react"
import NewRecipeForm from '../src/components/NewRecipeForm'
import RecipeInboxGrid from '../src/components/RecipeInboxGrid'
import GlobalState from '../src/context/GlobalState'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });



describe('<NewRecipeForm />', () => {
  const container = shallow(<NewRecipeForm />)
  
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('should have a field for the recipe title', () => {
    expect(container.find('#title'))
  })  

  it('should have a field for the recipe url', () => {
    expect(container.find('#link'))
  })
})