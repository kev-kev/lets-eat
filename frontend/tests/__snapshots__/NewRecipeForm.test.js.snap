// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewRecipeForm /> should match the snapshot 1`] = `"<main class=\\"MuiContainer-root MuiContainer-maxWidthXs\\"><div class=\\"makeStyles-paper-1\\"><h1 class=\\"MuiTypography-root MuiTypography-h5\\">submit a new recipe idea</h1><form class=\\"makeStyles-form-3\\" novalidate=\\"\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"title\\" id=\\"title-label\\">Recipe Title<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" autoComplete=\\"title\\" autofocus=\\"\\" id=\\"title\\" name=\\"title\\" required=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-8\\"><span>Recipe Title *</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"imgUrl\\" id=\\"imgUrl-label\\">Image URL</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"imgUrl\\" aria-invalid=\\"false\\" autoComplete=\\"current-imgUrl\\" id=\\"imgUrl\\" name=\\"imgUrl\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-8\\"><span>Image URL</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"link\\" id=\\"link-label\\">Link to recipe<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"link\\" aria-invalid=\\"false\\" autoComplete=\\"current-link\\" id=\\"link\\" name=\\"link\\" required=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-8\\"><span>Link to recipe *</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"notes\\" id=\\"notes-label\\">Notes</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline\\"><textarea aria-invalid=\\"false\\" autoComplete=\\"notes\\" id=\\"notes\\" name=\\"notes\\" rows=\\"4\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline\\"></textarea><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-8\\"><span>Notes</span></legend></fieldset></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">submit</span></button></form></div></main>"`;
